### PR TITLE
Td make num sum

### PR DESCRIPTION
**Pull request by tuanle618**
Checkout  num.sum.df in the code...
The current version might be a little to big for screen.. I included S3-print method to print NumSumObj.
Because of the percentiles the out.data.frame might be big.. 

NumSumObj mainly contains 3 elements. 

1. A num.sum.df as a whole data.frame for output (might be big)

2. A merged.list which contains each numeric variable (2.1) from the data.frame and the corresponding plot (2.2). Hence 2.1 is a replicate from 1, but I had to add 1 into the output as of convenience to print.

3. The NumSumTask

do you think the current summary is ok? Or might be too much?

### CODE REVIEW

File: R/getNumSum.R [hunk 1]
```diff
@@ -0,0 +1,36 @@
+getNumSum = function(data, features, target) {
+  num.data = subset(data, select = features)
+  #for the next calculations NAs will be removed
```
Comment (MiGraber) [inline R/getNumSum.R]: Instead of only mention it as comment, you could writing a warning message?

File: R/getNumSum.R [hunk 2]
```diff
@@ -0,0 +1,36 @@
+getNumSum = function(data, features, target) {
```
Comment (MiGraber) [inline R/getNumSum.R:21]: Even if this is a internal function, it would be nice to make a description in roxygenstyle. so you can avoid the `::` operator

**Comment by MiGraber** [comment]
Beside the notes it looks good, so merging should be fine.
I will try out the functions tomorrow (hopefully)

File: R/getNumSum.R (hunk 1 above)
Comment (tuanle618) [reply R/getNumSum.R]: alright will do that

File: R/getNumSum.R (hunk 2 above)
Comment (tuanle618) [reply R/getNumSum.R:21]: ok. will do it

**Comment by tuanle618** [comment]
@MiGraber @daryabusen : I changed the scripts. In addition I changed the input data.frame in plotFeatDistr() from num.data into data (the global one, containing all variables). I had to do it, because in case the target column is not-numeric (e.g factor) the function would have thrown an error, that target is not available in the num.data 

here the test code:  (target is factor)
```
data("diamonds", package = "ggplot2")
num.sum.task = makeNumSumTask(id = "Diamonds.Task", data = diamonds, target = "cut")
num.sum = makeNumSum(num.sum.task)
num.sum 
num.sum$num.sum$carat$plot
```

for numeric target:
```
data("Boston", package = "MASS") 
num.sum.task = makeNumSumTask(id = "Boston.Task", data = Boston, target = "medv") 
num.sum = makeNumSum(num.sum.task) 
num.sum 
num.sum$num.sum$age$plot
```

**Comment by MiGraber** [comment]
This looks quite nice. Maybe we should provide a description for the columns? mean, min, max, ... are clear but for example lower/upper bound arent without looking into the code. But this is low priority :)

**Comment by MiGraber** [comment]
getNumSum has no argument checks. Should we include them for intern function?

**Comment by tuanle618** [comment]
I will prepare the next days a description of the column. To the arg checking: I didnt include it, because the function is called within the `makeNumSum` function. There arg. checking takes place, but I will include it now. Double checking is always good. After that I will merge

**Pull request merged**